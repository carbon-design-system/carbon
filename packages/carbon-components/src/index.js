/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

// ====================//
// Imports and Exports //
// ====================//

// Base Elements & Components
// -------------
// - JavaScript classes for use with components and base-elements.
// - The following statements import classes from actual locations to
//   be consumed from this file instead of their actual locations.
export * from './globals/js/components';

export { default as settings } from './globals/js/settings';
