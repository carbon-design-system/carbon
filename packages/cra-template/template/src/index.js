import React from 'react';
import ReactDOM from 'react-dom';
import './index.scss';
import App from './App';
import reportWebVitals from './reportWebVitals';
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client';
const cache = new InMemoryCache();

const client = new ApolloClient({
  cache: cache,
  uri: 'https://api.github.com/graphql',
  headers: {
    authorization: `Bearer ${process.env.REACT_APP_GITHUB_PERSONAL_ACCESS_TOKEN}`,
  },
});

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <App />
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root')
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
