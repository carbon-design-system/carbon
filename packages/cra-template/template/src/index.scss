@use '@carbon/react';
@use '@carbon/react/scss/breakpoint' as *;
