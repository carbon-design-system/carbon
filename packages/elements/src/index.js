/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export {
  black,
  black100,
  blue,
  blue10,
  blue20,
  blue30,
  blue40,
  blue50,
  blue60,
  blue70,
  blue80,
  blue90,
  blue100,
  colors,
  coolGray,
  coolGray10,
  coolGray20,
  coolGray30,
  coolGray40,
  coolGray50,
  coolGray60,
  coolGray70,
  coolGray80,
  coolGray90,
  coolGray100,
  cyan,
  cyan10,
  cyan20,
  cyan30,
  cyan40,
  cyan50,
  cyan60,
  cyan70,
  cyan80,
  cyan90,
  cyan100,
  gray,
  gray10,
  gray20,
  gray30,
  gray40,
  gray50,
  gray60,
  gray70,
  gray80,
  gray90,
  gray100,
  green,
  green10,
  green20,
  green30,
  green40,
  green50,
  green60,
  green70,
  green80,
  green90,
  green100,
  magenta,
  magenta10,
  magenta20,
  magenta30,
  magenta40,
  magenta50,
  magenta60,
  magenta70,
  magenta80,
  magenta90,
  magenta100,
  orange,
  orange40,
  orange60,
  orange70,
  purple,
  purple10,
  purple20,
  purple30,
  purple40,
  purple50,
  purple60,
  purple70,
  purple80,
  purple90,
  purple100,
  red,
  red10,
  red20,
  red30,
  red40,
  red50,
  red60,
  red70,
  red80,
  red90,
  red100,
  rgba,
  teal,
  teal10,
  teal20,
  teal30,
  teal40,
  teal50,
  teal60,
  teal70,
  teal80,
  teal90,
  teal100,
  warmGray,
  warmGray10,
  warmGray20,
  warmGray30,
  warmGray40,
  warmGray50,
  warmGray60,
  warmGray70,
  warmGray80,
  warmGray90,
  warmGray100,
  white0,
  yellow,
  yellow20,
  yellow30,
  yellow40,
  yellow50,
} from '@carbon/colors';
export {
  fluid,
  fontFamilies,
  fontFamily,
  fontWeights,
  fontWeight,
  print,
  reset,
  getTypeSize,
  scale,
  styles,
  unstable_tokens,
} from '@carbon/type';
export {
  baseFontSize,
  rem,
  em,
  px,
  breakpoints,
  breakpointUp,
  breakpointDown,
  breakpoint,
  container,
  fluidSpacing,
  iconSize,
  miniUnit,
  miniUnits,
  layout,
  spacing,
} from '@carbon/layout';
export * from '@carbon/themes';
export {
  fast01,
  fast02,
  moderate01,
  moderate02,
  slow01,
  slow02,
  easings,
  motion,
} from '@carbon/motion';
