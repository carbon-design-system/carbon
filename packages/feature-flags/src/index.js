/**
 * Copyright IBM Corp. 2015, 2020
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { featureFlagInfo } from './generated/feature-flags';

const featureFlags = new Map();

for (let i = 0; i < featureFlagInfo.length; i++) {
  const featureFlag = featureFlagInfo[i];
  featureFlags.set(featureFlag.name, featureFlag.enabled);
}

/**
 * Check to see if a flag exists
 * @param {string} name
 */
function checkForFlag(name) {
  if (!featureFlags.has(name)) {
    throw new Error(`Unable to find a feature flag with the name \`${name}\``);
  }
}

/**
 * Add a feature flag
 * @param {string} name
 * @param {boolean} enabled
 */
export function add(name, enabled) {
  if (featureFlags.has(name)) {
    throw new Error(`The feature flag: ${name} already exists`);
  }
  featureFlags.set(name, enabled);
}

/**
 * Enable a feature flag
 * @param {string} name
 */
export function enable(name) {
  checkForFlag(name);
  featureFlags.set(name, true);
}

/**
 * Disable a feature flag
 * @param {string} name
 */
export function disable(name) {
  checkForFlag(name);
  featureFlags.set(name, false);
}

/**
 * Merge the given feature flags with the current set of feature flags.
 * Duplicate keys will be set to the value in the given feature flags.
 * @param {object} flags
 */
export function merge(flags) {
  Object.keys(flags).forEach((key) => {
    featureFlags.set(key, flags[key]);
  });
}

/**
 * Check if a feature flag is enabled
 * @param {string} name
 * @returns {boolean}
 */
export function enabled(name) {
  checkForFlag(name);
  return featureFlags.get(name);
}

export { featureFlagInfo as unstable_featureFlagInfo };
