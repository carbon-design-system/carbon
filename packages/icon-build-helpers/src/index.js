/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

'use strict';

const builders = require('./builders');
const Metadata = require('./metadata');
const Registry = require('./registry');

module.exports = {
  builders,
  Metadata,
  Registry,
};
