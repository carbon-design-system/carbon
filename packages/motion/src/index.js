/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const fast01 = '70ms';
export const fast02 = '110ms';
export const moderate01 = '150ms';
export const moderate02 = '240ms';
export const slow01 = '400ms';
export const slow02 = '700ms';

export const unstable_tokens = [
  'fast01',
  'fast02',
  'moderate01',
  'moderate02',
  'slow01',
  'slow02',
];

export const easings = {
  standard: {
    productive: 'cubic-bezier(0.2, 0, 0.38, 0.9)',
    expressive: 'cubic-bezier(0.4, 0.14, 0.3, 1)',
  },
  entrance: {
    productive: 'cubic-bezier(0, 0, 0.38, 0.9)',
    expressive: 'cubic-bezier(0, 0, 0.3, 1)',
  },
  exit: {
    productive: 'cubic-bezier(0.2, 0, 1, 0.9)',
    expressive: 'cubic-bezier(0.4, 0.14, 1, 1)',
  },
};

export function motion(name, mode) {
  if (!easings[name]) {
    throw new Error(
      `Unable to find easing \`${name}\` in our supported easings. Expected ` +
        `One of: ${Object.keys(easings).join(', ')}`
    );
  }

  const easing = easings[name];
  if (!easing[mode]) {
    throw new Error(
      `Unable to find a mode for the easing \`${name}\` called: \`${mode}\``
    );
  }

  return easing[mode];
}
