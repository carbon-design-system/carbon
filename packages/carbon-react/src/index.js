/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import './feature-flags';

export {
  Accordion,
  AccordionItem,
  AspectRatio,
  Breadcrumb,
  BreadcrumbItem,
  Button,
  ButtonSet,
  Checkbox,
  CodeSnippet,
  ComboBox,
  ComposedModal,
  ModalHeader,
  ModalBody,
  ModalFooter,
  ContentSwitcher,
  Copy,
  CopyButton,
  DangerButton,
  DataTable,
  Table,
  TableActionList,
  TableBatchAction,
  TableBatchActions,
  TableBody,
  TableCell,
  TableContainer,
  TableExpandHeader,
  TableExpandRow,
  TableExpandedRow,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
  TableToolbar,
  TableToolbarAction,
  TableToolbarContent,
  TableToolbarSearch,
  TableToolbarMenu,
  DatePicker,
  DatePickerInput,
  Dropdown,
  ErrorBoundary,
  ErrorBoundaryContext,
  FileUploader,
  Filename,
  FileUploaderButton,
  FileUploaderDropContainer,
  FileUploaderItem,
  Form,
  FluidForm,
  FormGroup,
  FormItem,
  FormLabel,
  Row,
  InlineLoading,
  Link,
  ListItem,
  Loading,
  Modal,
  MultiSelect,
  ToastNotification,
  InlineNotification,
  NotificationActionButton,
  NotificationButton,
  NotificationTextDetails,
  NumberInput,
  OrderedList,
  OverflowMenu,
  OverflowMenuItem,
  Pagination,
  PaginationNav,
  PrimaryButton,
  ProgressIndicator,
  ProgressStep,
  RadioButton,
  RadioButtonGroup,
  Search,
  SearchFilterButton,
  SearchLayoutButton,
  SecondaryButton,
  Select,
  SelectItem,
  SelectItemGroup,
  Switch,
  Slider,
  StructuredListWrapper,
  StructuredListHead,
  StructuredListBody,
  StructuredListRow,
  StructuredListInput,
  StructuredListCell,
  Tab,
  TabContent,
  Tabs,
  Tag,
  TextArea,
  TextInput,
  Tile,
  ClickableTile,
  SelectableTile,
  ExpandableTile,
  TileAboveTheFoldContent,
  TileBelowTheFoldContent,
  RadioTile,
  TileGroup,
  TimePicker,
  TimePickerSelect,
  Toggle,
  ToggleSmall,
  Toolbar,
  ToolbarItem,
  ToolbarTitle,
  ToolbarOption,
  ToolbarDivider,
  ToolbarSearch,
  Tooltip,
  TooltipDefinition,
  TooltipIcon,
  UnorderedList,
  SkeletonText,
  SkeletonPlaceholder,
  DataTableSkeleton,
  AccordionSkeleton,
  BreadcrumbSkeleton,
  ButtonSkeleton,
  CheckboxSkeleton,
  CodeSnippetSkeleton,
  DropdownSkeleton,
  FileUploaderSkeleton,
  NumberInputSkeleton,
  PaginationSkeleton,
  ProgressIndicatorSkeleton,
  RadioButtonSkeleton,
  SearchSkeleton,
  SelectSkeleton,
  SliderSkeleton,
  StructuredListSkeleton,
  TabsSkeleton,
  TagSkeleton,
  TextAreaSkeleton,
  TextInputSkeleton,
  ToggleSkeleton,
  ToggleSmallSkeleton,
  IconSkeleton,
  DatePickerSkeleton,
  Content,
  Header,
  HeaderContainer,
  HeaderGlobalAction,
  HeaderGlobalBar,
  HeaderMenu,
  HeaderMenuButton,
  HeaderMenuItem,
  HeaderName,
  HeaderNavigation,
  HeaderPanel,
  HeaderSideNavItems,
  Switcher,
  SwitcherItem,
  SwitcherDivider,
  SkipToContent,
  SideNav,
  SideNavDetails,
  SideNavDivider,
  SideNavFooter,
  SideNavHeader,
  SideNavIcon,
  SideNavItem,
  SideNavItems,
  SideNavLink,
  SideNavLinkText,
  SideNavMenu,
  SideNavMenuItem,
  SideNavSwitcher,
  // Experimental
  unstable_PageSelector,
  unstable_Pagination,
  unstable_TreeView,
  unstable_TreeNode,
  unstable_ContextMenu,
  unstable_ContextMenuDivider,
  unstable_ContextMenuGroup,
  unstable_ContextMenuItem,
  unstable_ContextMenuRadioGroup,
  unstable_ContextMenuSelectableItem,
  unstable_useContextMenu,
  unstable_Heading,
  unstable_Section,
} from 'carbon-components-react';

export {
  // Grid
  Grid,
  Column,
} from './components/Grid';
