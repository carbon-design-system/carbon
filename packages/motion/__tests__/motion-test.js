/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment node
 */

import { SassRenderer } from '@carbon/test-utils/scss';
import * as CarbonMotion from '../src';

const { render } = SassRenderer.create(__dirname);

describe('@carbon/motion', () => {
  test('Public API', () => {
    expect(CarbonMotion).toMatchSnapshot();
  });

  test('@carbon/motion/scss/motion.scss', async () => {
    const { getValue } = await render(`
      @use 'sass:meta';
      @use '../scss/motion';

      $_: get-value(meta.module-variables('motion'));
    `);
    const variables = getValue(0);
    expect(Object.keys(variables)).toMatchSnapshot();
  });
});
