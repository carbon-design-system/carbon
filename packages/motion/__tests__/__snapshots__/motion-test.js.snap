// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@carbon/motion @carbon/motion/scss/motion.scss 1`] = `
Array [
  "fast-01",
  "fast-02",
  "moderate-01",
  "moderate-02",
  "slow-01",
  "slow-02",
  "carbon--easings",
]
`;

exports[`@carbon/motion Public API 1`] = `
Object {
  "easings": Object {
    "entrance": Object {
      "expressive": "cubic-bezier(0, 0, 0.3, 1)",
      "productive": "cubic-bezier(0, 0, 0.38, 0.9)",
    },
    "exit": Object {
      "expressive": "cubic-bezier(0.4, 0.14, 1, 1)",
      "productive": "cubic-bezier(0.2, 0, 1, 0.9)",
    },
    "standard": Object {
      "expressive": "cubic-bezier(0.4, 0.14, 0.3, 1)",
      "productive": "cubic-bezier(0.2, 0, 0.38, 0.9)",
    },
  },
  "fast01": "70ms",
  "fast02": "110ms",
  "moderate01": "150ms",
  "moderate02": "240ms",
  "motion": [Function],
  "slow01": "400ms",
  "slow02": "700ms",
  "unstable_tokens": Array [
    "fast01",
    "fast02",
    "moderate01",
    "moderate02",
    "slow01",
    "slow02",
  ],
}
`;
